#!/usr/bin/env python3
PACKAGE = "strategy"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("game_start",    bool_t,    0, "A Boolen parameter",  False)
mode_enum = gen.enum([ gen.const("Setting",      str_t, "Setting",    "Setting the robot"),
                       gen.const("Navigating",   str_t, "Navigating", "the robot start to navagate")],
                       gen.const("Calculating",   str_t, "Calculating", "calculate the robot route")],
                     "An enum to set mode")
gen.add("Robot_mode", str_t, 0, "Select Game mode", "Setting", edit_method = mode_enum)

gen.add("get_loc",    bool_t,    0, "A Boolen parameter",  False)
# gen.add("game_start",    int_t,    0, "An Integer parameter", 50,  0, 100)
# gen.add("double_param", double_t, 0, "A double parameter",    .5, 0,   1)
# gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
# gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)


exit(gen.generate(PACKAGE, "strategy", "Robot"))

